packages/contracts/contracts/optimistic-ethereum/OVM/bridge/tokens/OVM_L2DepositedERC20.sol

```diff
@@ -3,65 +3,218 @@ pragma solidity >0.5.0 <0.8.0;
 pragma experimental ABIEncoderV2;
 
 /* Interface Imports */
-import { iOVM_L1TokenGateway } from "../../../iOVM/bridge/tokens/iOVM_L1TokenGateway.sol";
+import { iOVM_ERC20 } from "../../../iOVM/predeploys/iOVM_ERC20.sol";
+import { iOVM_L1ERC20Bridge } from "../../../iOVM/bridge/tokens/iOVM_L1ERC20Bridge.sol";
+import { iOVM_L2DepositedToken } from "../../../iOVM/bridge/tokens/iOVM_L2DepositedToken.sol";
+
+/* Library Imports */
+import { OVM_CrossDomainEnabled } from "../../../libraries/bridge/OVM_CrossDomainEnabled.sol";
 
 /* Contract Imports */
 import { UniswapV2ERC20 } from "../../../libraries/standards/UniswapV2ERC20.sol";
 
-/* Library Imports */
-import { Abs_L2DepositedToken } from "./Abs_L2DepositedToken.sol";
-
 /**
  * @title OVM_L2DepositedERC20
  * @dev The L2 Deposited ERC20 is an ERC20 implementation which represents L1 assets deposited into L2.
- * This contract mints new tokens when it hears about deposits into the L1 ERC20 gateway.
- * This contract also burns the tokens intended for withdrawal, informing the L1 gateway to release L1 funds.
- *
- * NOTE: This contract implements the Abs_L2DepositedToken contract using Uniswap's ERC20 as the implementation.
- * Alternative implementations can be used in this similar manner.
+ * This contract mints new tokens when it hears about deposits into the L1 ERC20 bridge.
+ * This contract also burns the tokens intended for withdrawal, informing the L1 bridge to release L1 funds.
+ * NOTE: This contract uses Uniswap's ERC20 as the implementation.
  *
  * Compiler used: optimistic-solc
  * Runtime target: OVM
  */
-contract OVM_L2DepositedERC20 is Abs_L2DepositedToken, UniswapV2ERC20 {
+contract OVM_L2DepositedERC20 is iOVM_L2DepositedToken, OVM_CrossDomainEnabled, UniswapV2ERC20 {
```

Comment:
I missed this before, but I was under then impression that there would also be a single standard bridge on L2. 
The token should be separate, and be mintable/burnable by the L2 Bridge.